frontend/src/components/TwoLevelDropdown.tsx

```diff
@@ -0,0 +1,295 @@
+}
+
+interface DropdownSubItem {
+  name: string;
+  secondaryName?: string;
+}
+
+export interface DropdownItem {
+  name: string;
+  subItems: DropdownSubItem[];
+}
+
+interface DropdownButtonProps {
+  dropdownRef: Ref<HTMLDivElement>;
+  toggleDropdown: (e: React.MouseEvent) => void;
+  selectedItem: SelectedItem;
+  title: string;
+}
+
+function DropdownButton(props: DropdownButtonProps) {
+  const { dropdownRef, toggleDropdown, selectedItem, title } = props;
+
+  const dropdownTooltip: string =
+    selectedItem.itemName && selectedItem.subItemName
+      ? `${selectedItem.itemName} > ${selectedItem.subItemName}` +
+        (selectedItem.subItemSecondaryName ? ` > ${selectedItem.subItemSecondaryName}` : '')
+      : title;
+  return (
+    <div className={classes(css.inlineContainer)} ref={dropdownRef}>
+      <Button onClick={toggleDropdown} className={classes(css.dropdown, css.textContainer)}>
+        <Tooltip title={dropdownTooltip} enterDelay={300} placement='top-start'>
+          <span className={classes(css.textContainer)}>
+            {selectedItem.itemName && selectedItem.subItemName ? (
+              <>
+                {selectedItem.itemName}
+                <ChevronRightSmallIcon />
+                {selectedItem.subItemName}
+                {selectedItem.subItemSecondaryName && (
+                  <>
+                    <ChevronRightSmallIcon />
+                    {selectedItem.subItemSecondaryName}
+                  </>
+                )}
+              </>
+            ) : (
+              title
+            )}
+          </span>
+        </Tooltip>
+        <ArrowDropDownIcon />
+      </Button>
+    </div>
+  );
+}
+
+interface DropdownSubMenuProps {
+  subDropdownActive: number;
+  setSubDropdownActive: (subDropdownActive: number) => void;
+  subDropdownIndex: number;
+  item: DropdownItem;
+  setSelectedItem: (selectedItem: SelectedItem) => void;
+  setDropdownActive: (dropdownActive: boolean) => void;
+}
+
+function DropdownSubMenu(props: DropdownSubMenuProps) {
+  const {
+    subDropdownActive,
+    setSubDropdownActive,
+    subDropdownIndex,
+    item,
+    setSelectedItem,
+    setDropdownActive,
+  } = props;
+
+  if (item.subItems.length === 0 || subDropdownActive !== subDropdownIndex) {
+    return <></>;
+  }
+
+  return (
+    <ul className={classes(css.dropdownSubmenu, css.dropdownMenu)}>
+      {item.subItems.map((subItem, subIndex) => (
+        <li
+          className={classes(css.dropdownElement)}
+          key={subIndex}
+          onClick={() => {
+            setSelectedItem({
+              itemName: item.name,
+              subItemName: subItem.name,
+              subItemSecondaryName: subItem.secondaryName,
+            });
+            setDropdownActive(false);
+            setSubDropdownActive(-1);
+          }}
+        >
+          <Tooltip
+            title={subItem.name + (subItem.secondaryName ? ` > ${subItem.secondaryName}` : '')}
+            enterDelay={300}
```

Comment:
For my education, what's the purpose of adding delay here?

Reply:
Adding the delay prevents the tooltip from appearing immediately if your cursor passes by the dropdown item. I added this `enterDelay` and the value of `300` since that seemed to be the standard across the KFP frontend: https://github.com/kubeflow/pipelines/search?q=enterDelay.

This delay also exists on the `RunList.tsx` component; I'm happy to remove it if you think it hinders UX.